charts/osm-arc/templates/validate-single-mesh.yml

```diff
@@ -0,0 +1,154 @@
+# 
+# The job in this YAML file verifies if the current k8s distribution has been tested for osm-arc. 
+# The job is executed in helm pre-install phase of OSM release lifecycle. 
+#
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: validate-single-mesh-account
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook-weight": "5"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: validate-single-mesh-role
+  annotations:
+    "helm.sh/hook-weight": "10"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+rules:
+- apiGroups: ["", "apps"]
+  resources: ["namespaces", "pods", "deployments", "secrets"]
+  verbs: ["get", "list"]
+{{- if eq .Values.Azure.Cluster.Distribution "openshift" }}
+- apiGroups:
+  - security.openshift.io
+  resourceNames:
+    - anyuid
+  resources:
+    - securitycontextconstraints
+  verbs:
+    - use
+{{- end }}
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: validate-single-mesh-role-binding
+  annotations:
+    "helm.sh/hook-weight": "15"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+subjects:
+- kind: ServiceAccount
+  name: validate-single-mesh-account
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: validate-single-mesh-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: validate-single-mesh
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook-weight": "20"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeededd
+spec:
+  ttlSecondsAfterFinished: 0
+  backoffLimit: 5
+  template:
+    metadata:
+      labels:
+        app: validate-single-mesh
+    spec:
+      serviceAccountName: validate-single-mesh-account
+      automountServiceAccountToken: true
+      restartPolicy: Never
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: validate-single-mesh-aks-addon
+          image: {{ .Values.kubectl.image.name }}:{{ .Values.kubectl.image.tag }}
+          imagePullPolicy: IfNotPresent
+          env: 
+            - name: enforcesinglemesh
+              value: {{ .Values.osm.OpenServiceMesh.enforceSingleMesh | quote }}
+            - name: validatenamespaces
+              value: {{ .Values.OpenServiceMesh.validateNamespaces }}
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              osm_namespaces=$(kubectl get deployments -l app=osm-controller -A -o=jsonpath={..metadata.namespace})
+              for namespace in $osm_namespaces
+              do
+                for validateNamespace in ${validatenamespaces} 
+                do
+                  if [ "$namespace" = "kube-system" ]; then
```

Comment:
Could this aks logic be bundled with the oss logic? Since the only differentiator here is that aks installs in kube-system?

Reply:
With the AKS add-on, you cannot see the helm release, as I believe the add-on is not installed via helm like OSS and Arc are. Thus the AKS-addon scenario needs to be handled differently from the OSS logic, where we need to use helm release data to distinguish between OSS and osm-arc releases (explained below). I thought it would make sense to do the OSS job in a separate container with the alpine-helm container image.

Reply:
Actually, I think there is a way to do this kubectl without needing to use a separate container with helm. Will explore this.

Reply:
Ok, I remember why I decided to do this via helm CLI. For OSS, the kubect metadata via `kubectl get deployment -l app=osm-controller -o json` doesn't include the "enforceSingleMesh" field, so the only way to access it is by parsing the values via `helm get values`.

Reply:
https://github.com/openservicemesh/osm/blob/dde84836e54eef62aaf7688b9f2d1c4dd730d366/charts/osm/templates/osm-deployment.yaml#L10

The enforceSingleMesh is added as a label, I think you can use this with the `-l` flag?

Reply:
Found another issue with trying to use kubectl, separate from enforceSingleMesh. 

Kubectl is able to parse the helm release name, but not the actual chart name. The chart name is the same across releases (either osm or osm-arc), thus this is what I use to distinguish between osm-arc and osm releases. I believe this can only be accessed via the helm CLI.